aspnetcore/blazor/Security-guidance.md

```diff
@@ -0,0 +1,254 @@
+* It is also recommended that you avoid including script tags as part of your component render tree for the reasons stated above.
+
+As part of protecting against XSS attacks we recommend looking into application specific mitigations like content security policy (CSP). See https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP for more details.
+
+## Cross-origin protection
+Cross-origin attacks involve a client from a different origin performing some action against your server. This action is typically a form POST (CSRF), but opening a websocket is also possible. Blazor server-side applications offer the same guarantees that any other SignalR application using the hub protocol offers.
+* A browser from a different origin can't connect to the blazor-server side application and start a new session (circuit) with the default application configuration.
+  * This is due to CORS being disabled by default and SignalR forcing a preflight request against the server as part of the hub protocol handshake.
+* If you enable CORS in your application then you might need to take extra steps to protect your application depending on your CORS configuration.
+  * If you enable CORS globally you can disable it specifically for the blazor server-side hub by adding the `DisableCorsAttribute` metadata to the endpoint metadata after calling `hub.MapBlazorHub()`.
+
+## Click-jacking
+Click-jacking involves rendering a site as an iframe inside a site from a different origin in order to trick the user into performing actions on the site under attack.
+
+If you want to protect your application from being rendered inside an iframe we recommend using CSP and the `X-Frame-Options` header. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options for more details
+
+## Open redirections
+When a blazor server-side application session starts the server performs some validation against the urls sent as part of starting the session.
+
+When a user clicks a link on the client, the url for the link is sent to the server which decides what to do with it, for example perform a client-side navigation or indicate the browser to go to the new location.
+
+Components can also trigger navigation requests programatically through the user of NavigationManager. In such cases, the application might perform a client-side navigation or indicate the browser to go to the new location.
+
+For that reason components must avoid using user input as part of the navigation call arguments and validate such arguments to ensure that the target is allowed by the application.
+
+Otherwise a malicious user could force the browser to go to an attacker controller site by tricking the application into using some user input as part of the invocation to Navigate.
+
+This advice also applies when rendering links as part of the application. Always use relative links if possible and validate that absolute links destinations are valid before including them in the page.
```

Comment:
It feels like we've gone way off track from Blazor-specific guidance. This applies to literally any web application.

Might it be worth taking a pass through this whole doc and trying to prune out anything that's not specific to Blazor? We don't want to be responsible for maintaining a general guide to web security.